src/main/kotlin/blackjack/domain/player/Dealer.kt

```diff
@@ -0,0 +1,46 @@
+package blackjack.domain.player
+
+import blackjack.domain.InputInterface
+import blackjack.domain.Money
+import blackjack.domain.OutputInterface
+import blackjack.domain.card.Card
+import blackjack.domain.card.Deck
+
+/**
+ * 딜러
+ * Created by Jaesungchi on 2022.06.15..
+ */
+class Dealer(initCards: List<Card>) : Player(DEALER_NAME, initCards) {
+    fun getBatResult(users: List<User>): Money {
+        var money = Money()
+        users.forEach {
+            money += matchUser(it)
+        }
+        return money
+    }
+
+    override fun hitStage(deck: Deck, input: InputInterface, output: OutputInterface) {
+        while (!isOverHitScore()) {
+            output.drawDealerHitMessage()
+            hit(deck.takeCard())
+        }
+    }
+
+    private fun matchUser(user: User): Money {
+        return when (match(user)) {
+            Match.WIN, Match.WIN_BLACKJACK -> user.money
+            Match.LOSE -> Money(-user.money.value)
+            Match.LOSE_BLACKJACK -> Money(-user.money.times(BLACKJACK_WIN_PROFIT_MARGIN).value)
+            else -> Money()
+        }
+    }
+
+    private fun isOverHitScore(): Boolean {
+        return cards.getScore().value >= HIT_SCORE
```

Comment:
```suggestion
        return cards.getScore() >= HIT_SCORE // HIT_SCORE 는 Score 타입
```
살짝 과할수도 있지만 객체에게 메시지를 보내는 것이므로 좀 더 객체지향적이라고 생각해요.